resources/schema.py

```diff
@@ -125,11 +126,25 @@ class Meta:
     mooring_type = BerthMooringTypeEnum(required=True)
 
 
+class BerthNodeFilterSet(django_filters.FilterSet):
+    min_width = django_filters.NumberFilter(
+        field_name="berth_type__width", lookup_expr="gte"
+    )
+    min_length = django_filters.NumberFilter(
+        field_name="berth_type__length", lookup_expr="gte"
+    )
+
+    class Meta:
+        model = Berth
+        fields = ["min_width", "min_length"]
+
+
 class BerthNode(DjangoObjectType):
     class Meta:
         model = Berth
         fields = ("id", "number", "pier", "berth_type", "comment")
         interfaces = (relay.Node,)
+        filterset_class = BerthNodeFilterSet
```

Comment:
again, uniformity issue: [here](https://github.com/City-of-Helsinki/berth-reservations/blob/develop/applications/new_schema.py#L122) we inject the filterset at the ConnectionField level and it kinda makes sense that way, since it is more related to this specific connection than to the Node.

Reply:
That was my first thought as well. However, to be able to use the filter on the node (not query) level, it has to be injected on the node class.

If we inject it on the [`Query`](https://github.com/City-of-Helsinki/berth-reservations/pull/144/files#diff-09f1721fc193a5a442fee1cf24f5bd49R647) level, it will only allow using the filter if the root query is `berths`.

Reply:
good catch 👍

Reply:
Having the `FilterSet` on the `Node` level allows to do both following queries without modifiying the parent (`piers`) node.

```graphql
# (showcased irl)
query Berths {
  berths(minWidth: 7, minLength: 10) {
    edges {
      node {
        id
        berthType {
          id
          width
          length
        }
      }
    }
  }
}

query Piers {
  piers {
    edges {
      node {
        id
        properties {
          berths(minWidth: 7, minLength: 10) {
            edges {
              node {
                id
                berthType {
                  id
                  width
                  length
                }
              }
            }
          }
        }
      }
    }
  }
}```